### PR TITLE
DR-774 Ready for test flows

### PR DESCRIPTION
**Pull request by jope1**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/actions/checkout-repository/action.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ inputs:
   management-tag:
     description: "Version of the uec-dos-management repo to checkout"
     required: false
-    default: v0.5
+    default: task/DR-774_Ready_for_test_flows
```
Comment (timrickwood) [inline .github/actions/checkout-repository/action.yaml:16]: This will need to change

File: .github/workflows/generate-tag-name.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    runs-on: ubuntu-latest
+    timeout-minutes: 2
+    outputs:
+      tag_name: ${{ steps.generate-tag-name.outputs.tag_name }}
+    steps:
+      - name: "Checkout code"
+        uses: NHSDigital/uec-dos-management/.github/actions/checkout-repository@task/DR-774_Ready_for_test_flows
+      - name: "Generate tag name"
+        id: "generate-tag-name"
+        uses: NHSDigital/uec-dos-management/.github/actions/generate-tag-name@task/DR-774_Ready_for_test_flows
```
Comment (timrickwood) [inline .github/workflows/generate-tag-name.yaml:35]: branch ref will need to change

File: .github/workflows/tag-repository.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        default: ""
+        type: string
+jobs:
+  tag_repository:
+    name: "Tag repository"
+    runs-on: ubuntu-latest
+    timeout-minutes: 2
+    steps:
+      - name: "Checkout code"
+        uses: NHSDigital/uec-dos-management/.github/actions/checkout-repository@task/DR-774_Ready_for_test_flows
```
Comment (timrickwood) [inline .github/workflows/tag-repository.yaml:27]: branch name change

File: .github/workflows/test-ready-for-test-label.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        tag-repository,
+        deploy-infrastructure
+      ]
+    uses: ./.github/workflows/build-and-deploy-application.yaml
+    with:
+      environment: mgmt
+      workspace: ${{ needs.metadata.outputs.workspace }}
+      services: "['healthcare-services-data-manager']"
+      tag: ${{ needs.tag-repository.outputs.tag }}
+    secrets: inherit
```
Comment (timrickwood) [inline .github/workflows/test-ready-for-test-label.yaml:74]: Do we want to tear it down at the end as well ?

File: .github/actions/checkout-repository/action.yaml (hunk 1 above)
Comment (jope1) [reply .github/actions/checkout-repository/action.yaml]: Done

File: .github/workflows/generate-tag-name.yaml (hunk 2 above)
Comment (jope1) [reply .github/workflows/generate-tag-name.yaml]: Done

File: .github/workflows/tag-repository.yaml (hunk 3 above)
Comment (jope1) [reply .github/workflows/tag-repository.yaml]: Done

File: .github/workflows/test-ready-for-test-label.yaml (hunk 4 above)
Comment (jope1) [reply .github/workflows/test-ready-for-test-label.yaml:74]: Added

**Review by timrickwood — APPROVED** [review]

**Pull request merged**